File: blnc/blnc.lua
local families, giveup, ratio, exct=100000, 100, 515, 1000
local boys, girls, children=0, 0, 0
local j=0

if #arg>=1 then giveup=arg[1]+0 end

math.randomseed(os.time())

for i=1, families do
	while (boys==0 or girls==0 or boys~=girls) and (boys+girls<giveup) do
		if math.random(1, exct)<=ratio then boys=boys+1 else girls=girls+1 end
	end
	if boys==girls then children=children+boys+girls j=j+1 end
	boys, girls=0, 0
end

print("families: " .. families .. ",", "give up at: " .. giveup .. ",", "sex ratio: " .. ratio .. ":" .. (exct-ratio))
print("child avg: " .. (children/families), "give up: " .. ((1-j/families)*100) .. "%")

File: blnc/blnc.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

const int FAMILIES=1000000;

int main(int argc, char** argv)
{
	int i, j;
	unsigned int c;
	unsigned long long boys=0, girls=0, children=0;

	FILE* urandom=fopen("/dev/urandom", "r");
	srandom((unsigned)time(NULL)+fgetc(urandom));
	fclose(urandom);

	for(i=0; i<FAMILIES; i++)
	{
		while(boys==0||girls==0||boys!=girls)
		{
			if(j==0) c=random();
			if(c&1) boys++;
			else girls++;
			c>>=1;
			j=(j+1)%31;
		}
		children+=boys+girls;
		boys=girls=0;
	}

	printf("%.4f\n", ((double)children)/((double)FAMILIES));

	return 0;
}
